### PR TITLE
Features/on incomming message callback #13

### LINKED ISSUE #13: onIncommingMessage callback
Create a callback for incomming message

### PR DESCRIPTION
**Pull request by hubertusputu1**
## Context
#13 

## Features
- [x] Add onIncommingMessage callback which return all the messages

### CODE REVIEW

File: src/style.less [hunk 1]
```diff
@@ -49,6 +49,26 @@
     display: grid;
     grid-template-rows: 1fr auto;
 
+    [data-badge] {
```
Comment (npaton) [inline src/style.less]: Is this still being used ?

File: src/Messages.js [hunk 2]
```diff
@@ -27,12 +27,16 @@ class Messages extends React.PureComponent {
...
     const { messages: prevMessages } = prevProps;
-    const { messages: currentMessages } = this.props;
+    const { messages: currentMessages, onIncommingMessage } = this.props;
 
     if (
       this.messagesEl.current !== null &&
       currentMessages.length > prevMessages.length
     ) {
+      if (onIncommingMessage) {
+        onIncommingMessage(currentMessages);
```
Comment (npaton) [inline src/Messages.js]: Could we send only new messages?

**Review by npaton — CHANGES_REQUESTED** [review]
Looks good, only a couple of minor changes.

File: src/style.less (hunk 1 above)
Comment (hubertusputu1) [reply src/style.less]: Sorry it's not, i will remove it.

File: src/Messages.js (hunk 2 above)
Comment (hubertusputu1) [reply src/Messages.js]: Yes we can.

File: src/Chat.js [hunk 3]
```diff
@@ -17,7 +16,7 @@ export default class Chat extends React.PureComponent {
     const { onNewMessage, scope, customKey } = this.props;
 
     if (onNewMessage) {
-      msg = onNewMessage(msg);
+      msg = onNewMessage(msg, this.state.unreadMsg);
```
Comment (npaton) [inline src/Chat.js]: Is there still an unreadMsg count here?

**Review by npaton — APPROVED** [review]

**Pull request merged**